src/lib/operators/alias_operator.cpp

```diff
@@ -79,12 +79,17 @@ std::shared_ptr<const Table> AliasOperator::_on_execute() {
       auto sort_definitions = std::vector<SortColumnDefinition>{};
       sort_definitions.reserve(sorted_by.size());
 
-      // Adapt column ids
+      // Adapt column ids of chunk sort definitions
       for (const auto& sort_definition : sorted_by) {
+        // In some edge cases an input table might be sorted by a column that is not included in the list of columns.
+        // This can happen when an expression occurs repeatedly (e.g., `SELECT a as a1, a as a2`) and the LQP
+        // translator references the first occurrence twice (leaving the second (sorted) occurrence unreferenced, see
+        // issue #2321 for more details). We thus check, if the sorted column is part of the input table.
         const auto it = std::find(_column_ids.cbegin(), _column_ids.cend(), sort_definition.column);
-        Assert(it != _column_ids.cend(), "Chunk is sorted by an invalid column ID.");
-        const auto index = ColumnID{static_cast<uint16_t>(std::distance(_column_ids.cbegin(), it))};
-        sort_definitions.emplace_back(SortColumnDefinition(index, sort_definition.sort_mode));
+        if (it != _column_ids.cend()) {
+          const auto index = ColumnID{static_cast<uint16_t>(std::distance(_column_ids.cbegin(), it))};
+          sort_definitions.emplace_back(SortColumnDefinition(index, sort_definition.sort_mode));
+        }
```

Comment:
Assert that the number of output sort definitions matches that of input definitions?

Reply:
It does not equal. Since the alias operator never sees the second sorted columns (it's removed in the LQP translator), we lose the second sorted column.

We could iterate the columns and check if they are part of the sorted definition instead of the current way of iterating the sorted information. This way, we should have the same length.

Is this was you recommend?

Reply:
> we lose the second sorted column

Right, but won't we find the first column twice?

Reply:
Changed the code to find the first column twice, added an assert.